pkg/model/evaluation.go

```diff
@@ -0,0 +1,109 @@
+package model
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/lib/math"
+)
+
+const (
+	EvalStatusBlocked   = "blocked"
+	EvalStatusPending   = "pending"
+	EvalStatusComplete  = "complete"
+	EvalStatusFailed    = "failed"
+	EvalStatusCancelled = "canceled"
+)
+
+// Evaluation is just to ask the scheduler to reassess if additional job instances must be
+// scheduled or if existing ones must be stopped.
+// It is possible that no action is required if the scheduler sees the desired job state matches the observed state.
+// This allows the triggers (e.g. APIs, Node Manager) to be lightweight and submit evaluation on state changes without
+// having to do complex logic to decide what actions to take.
+type Evaluation struct {
+	// ID is the unique identifier of the evaluation.
+	ID string
+
+	// Namespace is the namespace the evaluation is created in
```

Comment:
I may have missed it, but what sort of namespaces are you thinking we might have?  Are these namespaces supposed to identify other requester nodes?

Reply:
namespaces should be used to identify workloads for different users and organizations. We don't have those yet, but we should slowly introduce them with the upcoming refactoring. This should allow a single cluster to be used by different users or organizations. Auth should also consider namespaces when authorizing user actions.

Today you can think of `clientid` as our namespace where we allow users to list jobs only matching their client.